src/Serilog.Sinks.AzureDataExplorer/Durable/BookmarkFile.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Text;
+
+namespace Serilog.Sinks.AzureDataExplorer.Durable
+{
+    /// <summary>
+    /// https://github.com/serilog/serilog-sinks-seq/blob/v4.0.0/src/Serilog.Sinks.Seq/Sinks/Seq/Durable/BookmarkFile.cs
+    /// this class is a file-based bookmark mechanism.
+    /// It provides a way to persist a bookmark across multiple executions of an application.
+    /// The bookmark is stored in a file and is represented as a FileSetPosition object, which contains the starting position of the next line in a file and the file name.
+    /// The class provides methods to read and write the bookmark to the file, and implements the IDisposable interface to clean up any resources it uses.
+    /// </summary>
+    sealed class BookmarkFile : IDisposable
+    {
+        readonly FileStream m_bookmark;
+
+        public BookmarkFile(string bookmarkFilename)
+        {
+            m_bookmark = System.IO.File.Open(bookmarkFilename, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read);
+        }
+
+        public FileSetPosition TryReadBookmark()
+        {
+            if (m_bookmark.Length != 0)
```

Comment:
m_bookmark**_?_** will make it safe against null values (the question mark). Please check

Reply:
This m_bookmark doesn't seem to null anytime. Since it is initialized via the constructor. Even the caller to BookMarkFile -> FileSet is also initialized with fileName during LogShipper constructor invocation.